packages/pieces/community/google-drive/src/lib/action/set-public-access.ts

```diff
@@ -0,0 +1,40 @@
+import { Property, createAction } from '@activepieces/pieces-framework';
+import { googleDriveAuth } from '../../';
+import { google } from 'googleapis';
+import { OAuth2Client } from 'googleapis-common';
+
+export const setPublicAccess = createAction({
+  auth: googleDriveAuth,
+  name: 'set_public_access',
+  description: 'Set public access for a file or folder',
+  displayName: 'Set public access',
+  props: {
+    fileId: Property.ShortText({
+      displayName: 'File or Folder ID',
+      description: 'The ID of the file or folder to update permissions for',
+      required: true,
+    }),
+  },
+  async run(context) {
+    const authClient = new OAuth2Client();
+    authClient.setCredentials(context.auth);
+
+    const fileId = context.propsValue.fileId;
+
+    const drive = google.drive({ version: 'v3', auth: authClient });
+    const permission = {
+      role: 'reader',
+      type: 'anyone',
+    };
+    const res = await drive.permissions.create({
+      fileId: fileId,
+      requestBody: permission,
+    });
+
+    const file = await drive.files.get({
+      fileId: fileId,
+      fields: 'webContentLink',
```

Comment:
@pfernandez98 The files.get response doesn't contain any webContentLink. Is it necessary in action output?

Reply:
Actually, It's needed for the "downloadUrl" field I return with the response @kishanprmr